File: javascript/6_kyu/delete_occurrences_of_an_element.js
// Delete Occurrences of an Element... More than N Times
// https://www.codewars.com/kata/554ca54ffa7d91b236000023/train/javascript

function count(array, num) {
    var count = 0;
    for (var i = 0; i < array.length; i++) {
        if ( array[i] == num ) {
            count++;
        }
    }
    return count;
}

function deleteNth(seq, n) {
    var ans = [];

    for (var i = 0; i < seq.length; i++) {
        var occurrence = count(ans, seq[i]);

        if ( occurrence < n ) {
            ans.push(seq[i]);
        }
    }

    return ans;
}

console.log(deleteNth([20,37,20,21], 1));
console.log(deleteNth([1,1,3,3,7,2,2,2,2], 3));

File: python/6_kyu/delete_occurrences_of_an_element.py
# Delete Occurrences of an Element... More than N Times
# https://www.codewars.com/kata/554ca54ffa7d91b236000023/train/python

def delete_nth(seq, n):
    ans = []

    for i in seq:
        if ans.count(i) < n:
            ans.append(i)
        else:
            continue
    
    return ans

print(delete_nth([20,37,20,21], 1))
print(delete_nth([1,1,3,3,7,2,2,2,2], 3))
print(delete_nth([1,1,1,1], 2))